### PR TITLE
feature/temp-cache-last-processed-file-to-avoid-repeat-processing

### PR DESCRIPTION
**Pull request by chopkinsmade**
### Description of change
Use the IngestedFile table for tracking contact consent files that are ingested

### Checklist

* [ ] Has this branch been rebased on top of the current `main` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [ ] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/main/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

File: datahub/company/tasks/contact.py [hunk 1]
```diff
@@ -206,8 +208,16 @@ def ingest(self):
...
+        if IngestedFile.objects.filter(filepath=file_key).exists():
+            logger.info(
+                'File %s has already been processed',
+                file_key,
+            )
+            return
+
         try:
             self.sync_file_with_database(s3_client, file_key)
+            IngestedFile.objects.create(filepath=file_key)
```
Comment (baarkerlounger) [inline datahub/company/tasks/contact.py:220]: Does `file_key` here include the prefix? i.e. the full S3 path?

If not you could potentially get clashes with other ingestion file sources?

File: datahub/company/test/tasks/test_contact_task.py [hunk 2]
```diff
@@ -33,6 +33,8 @@
     schedule_update_contact_consent,
 )
 from datahub.company.test.factories import CompanyFactory, ContactFactory
+from datahub.company_activity.models.ingested_file import IngestedFile
```
Comment (baarkerlounger) [inline datahub/company/test/tasks/test_contact_task.py:36]: Looks like the refactor has been merged now, so could probably use the generic one now

File: datahub/company/tasks/contact.py (hunk 1 above)
Comment (chopkinsmade) [reply datahub/company/tasks/contact.py:220]: Yeh its the full key minus the bucket name, so for example `data-flow/exports/MergeConsentsPipeline/20241211T143556.jsonl.gz`

File: datahub/company/test/tasks/test_contact_task.py (hunk 2 above)
Comment (chopkinsmade) [reply datahub/company/test/tasks/test_contact_task.py]: Switched that over to the generic table

**Review by DeanElliott96 — APPROVED** [review]

**Pull request merged**